port/unix/omrsignal.c

```diff
@@ -1795,3 +1807,37 @@ omrsig_chain_at_shutdown_and_exit(struct OMRPortLibrary *portLibrary)
 }
 #endif /* defined(OMRPORT_OMRSIG_SUPPORT) */
 
+/**
+ * This function will unblock signals listed in signalMap by changing the
+ * signal mask of the calling thread. This function should only be called
+ * during initialization from the main thread and before creation of other
+ * threads.
+ *
+ * @param[in] void
+ *
+ * @return TRUE on success and FALSE on failure
+ */
+static BOOLEAN
+unblockSignals(void) {
+	sigset_t signalSet;
+	int i = 0;
+
+	if (0 != sigemptyset(&signalSet)) {
+		return FALSE;
+	}
+
+	/* Iterate through all signals listed in signalMap. */
+	for (i = 0; i < sizeof(signalMap)/sizeof(signalMap[0]); i++) {
+		/* Add the current signal to the signal set. */
+		if (0 != sigaddset(&signalSet, signalMap[i].unixSignalNo)) {
+			return FALSE;
+		}
+	}
+
+	/* Unblock the signals listed in signalMap. */
+	if (0 != sigprocmask(SIG_UNBLOCK, &signalSet, NULL)) {
+		return FALSE;
+	}
+
+	return TRUE;
```

Comment:
Can you reformat this to have a single return point?

Reply:
Please also add some tracepoints to this function, particularly for the error cases.

Reply:
Tracepoints won't work in initialization code. Asserts are used instead of tracepoints at such locations. I will add asserts.

Reply:
I'm not sure asserts help here.  The goal is to make it easy to understand what failed and why - not to take the runtime down quicker :)